lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.to_sorted.js

```diff
@@ -0,0 +1,211 @@
+*
+* @private
+* @param {number} a - first value for comparison
+* @param {number} b - second value for comparison
+* @returns {number} comparison result
+*/
+function compareFcn( a, b ) {
+	if ( a > b ) {
+		return 1;
+	}
+	if ( a < b) {
+		return -1;
+	}
+	return 0;
+}
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof factory, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns a function', function test( t ) {
+	var ctor = factory( 'float64' );
+	t.strictEqual( isFunction( ctor ), true, 'returns expected value' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the returned function is an `toSorted` method', function test( t ) {
+	var ctor = factory( 'float64' );
+	t.strictEqual( hasOwnProp( ctor.prototype, 'toSorted' ), true, 'returns expected value' );
+	t.strictEqual( isFunction( ctor.prototype.toSorted ), true, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a typed array instance', function test( t ) {
+	var values;
+	var ctor;
+	var arr;
+	var i;
+
+	ctor = factory( 'float64' );
+	arr = new ctor( 'little-endian', [ 1.0, 2.0, 3.0, 4.0, 5.0 ] );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.toSorted.call( value, compareFcn );
+		};
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var ctor;
+	var arr;
+	var i;
+
+	ctor = factory( 'float64' );
+	arr = new ctor( 'little-endian', [ 1.0, 2.0, 3.0, 4.0 ] );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.toSorted( value );
+		};
+	}
+});
+
+tape( 'the method returns an empty array if operating on an empty array', function test( t ) {
+	var ctor;
+	var out;
+	var arr;
+
+	ctor = factory( 'float64' );
+	arr = new ctor( 'little-endian' );
+	out = arr.toSorted( compareFcn );
+
+	t.strictEqual( out.length, 0, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method returns a new typed array containing elements in ascending order', function test( t ) {
+	var expected;
+	var values;
+	var ctor;
+	var arr;
+	var out;
+	var i;
+
+	ctor = factory( 'float64' );
+	arr = new ctor( 'little-endian', [ 5.0, 4.0, 1.0, 2.0, 3.0 ] );
+	expected = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	out = arr.toSorted( compareFcn );
+	values = [];
+	fcn();
```

Comment:
Rather than use `fcn` to generate a values array, just use `array/base/assert/has-same-values` and compare to an expected `Float64ArrayFE`.